distributed/protocol/serialize.py

```diff
@@ -817,11 +817,12 @@ def _is_msgpack_serializable(v):
         v is None
         or typ is str
         or typ is bool
+        or typ is bytes
         or typ is int
         or typ is float
         or isinstance(v, dict)
         and all(map(_is_msgpack_serializable, v.values()))
-        and all(typ is str for x in v.keys())
+        and all(isinstance(x, str) for x in v.keys())
```

Comment:
is there a reason for not opting for:

```suggestion
        and all(type(x) is str for x in v.keys())
```

Reply:
no particular reason

Reply:
with `strict_types` it's not possible to serialize str subclasses

```python
>>> class Foo(str): pass
... 
>>> msgpack.dumps({Foo("x"): "y"}, use_bin_type=True, strict_types=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/graingert/anaconda3/envs/dask-distributed/lib/python3.10/site-packages/msgpack/__init__.py", line 38, in packb
    return Packer(**kwargs).pack(o)
  File "msgpack/_packer.pyx", line 294, in msgpack._cmsgpack.Packer.pack
  File "msgpack/_packer.pyx", line 300, in msgpack._cmsgpack.Packer.pack
  File "msgpack/_packer.pyx", line 297, in msgpack._cmsgpack.Packer.pack
  File "msgpack/_packer.pyx", line 229, in msgpack._cmsgpack.Packer._pack
  File "msgpack/_packer.pyx", line 291, in msgpack._cmsgpack.Packer._pack
TypeError: can not serialize 'Foo' object
>>> 
```